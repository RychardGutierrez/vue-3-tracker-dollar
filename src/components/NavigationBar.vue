<template>
  <header class="sticky top-0 bg-white shadow-sm">
    <nav
      class="container flex flex-col sm:flex-row items-center gap-4 text-black py-6 px-0"
    >
      <RouterLink :to="routerHome">
        <div class="flex items-center gap-3">
          <img :src="iconDollar" alt="dollar" class="w-11" />
          <p class="text-2xl">Dollar Tracker</p>
        </div>
      </RouterLink>
      <div class="flex gap-5 flex-1 justify-end">
        <img
          v-if="route.query.preview"
          :src="iconSave"
          alt="Save Country"
          class="w-9 hover:shadow-xl duration-150 cursor-pointer"
          @click="handleSaved"
        />

        <img
          @click="toogleModal"
          :src="iconHelp"
          alt="Help"
          class="w-9 hover:shadow-xl rounded-full duration-150 cursor-pointer"
        />
      </div>
    </nav>
  </header>
  <HelpModal :modal-active="modalActive" @close-modal="toogleModal" />
</template>

<script setup>
import { RouterLink, useRoute, useRouter } from 'vue-router';
import { ref } from 'vue';

import { useTrackerStore } from '@/store';
import { updatedStore } from '@/data/store';
import { currentDate } from '@/helper/dateTime';

import iconDollar from '@/assets/dollar.svg';
import iconHelp from '@/assets/help.svg';
import iconSave from '@/assets/save.svg';
import HelpModal from '@/components/HelpModal.vue';

const route = useRoute();
const router = useRouter();
const modalActive = ref(false);
const savedCities = ref([]);

const trackerStore = useTrackerStore();

const routerHome = { name: 'home' };

const toogleModal = () => {
  modalActive.value = !modalActive.value;
};

const handleSaved = () => {
  const trackerObjec = {
    country: route.params.country,
    official: route.query.official,
    symbol: route.query.symbols,
    savedDate: currentDate(),
    rate: trackerStore.state.value,
  };
  const { id } = updatedStore(trackerObjec);

  let query = Object.assign({}, route.query);
  delete query.preview;
  query.id = id;
  router.replace({ query });
};
</script>

<style lang="scss" scoped></style>
