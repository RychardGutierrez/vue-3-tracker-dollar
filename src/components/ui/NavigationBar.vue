<template>
  <header class="sticky top-0 bg-white shadow-sm">
    <nav class="container flex flex-col sm:flex-row items-center gap-4 text-black py-6 px-0">
      <RouterLink :to="routerHome">
        <div class="flex items-center gap-3">
          <img :src="iconDollar" alt="dollar" class="w-11" />
          <p class="text-2xl">Dollar Tracker</p>
        </div>
      </RouterLink>
      <div class="flex gap-5 flex-1 justify-end">
        <img v-if="isPreview" :src="iconSave" alt="Save Country" class="w-9 hover:shadow-xl duration-150 cursor-pointer"
          @click="saveLocalStore" />

        <img @click="toogleModal" :src="iconHelp" alt="Help"
          class="w-9 hover:shadow-xl rounded-full duration-150 cursor-pointer" />
      </div>
    </nav>
  </header>
  <HelpModal :modal-active="modalActive" @close-modal="toogleModal" />
</template>

<script setup>
import { RouterLink } from 'vue-router';

import iconDollar from '../../assets/dollar.svg';
import iconHelp from '../../assets/help.svg';
import iconSave from '../../assets/save.svg';
import HelpModal from '../../components/ui/HelpModal.vue';
import useNavigationBar from '../../composables/useNavigationBar';

const routerHome = { name: 'home' };

const { saveLocalStore, toogleModal, isPreview, modalActive } = useNavigationBar();

</script>
