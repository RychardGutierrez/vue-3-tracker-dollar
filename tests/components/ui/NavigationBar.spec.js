import { expect } from '@jest/globals';
import { mount, shallowMount } from '@vue/test-utils';
import { nextTick } from 'vue';
import { getRouter } from 'vue-router-mock';

import NavigationBar from '../../../src/components/ui/NavigationBar.vue';

jest.mock('../../../src/assets/dollar.svg', () => 'dollar-icon');
jest.mock('../../../src/assets/help.svg', () => 'help-icon');
jest.mock('../../../src/assets/save.svg', () => 'save-icon');

const mockUseNavigationBar = {
  saveLocalStore: jest.fn(),
  toogleModal: jest.fn(),
  isPreview: false,
  modalActive: false,
};

jest.mock('../../../src/composables/useNavigationBar', () => {
  const useNavigationBar = () => mockUseNavigationBar;
  return useNavigationBar;
});

describe('NavigationBar component', () => {
  let wrapper = null;

  beforeEach(() => {
    wrapper = shallowMount(NavigationBar, { stubs: ['router-link'] });
  });

  afterEach(() => {
    jest.clearAllMocks();
  });

  test('should to match with snapshot', () => {
    expect(wrapper.html()).toMatchSnapshot();
  });

  test('should call toogleModal method when help is clicked', async () => {
    const helpModal = wrapper.find('[alt="Help"]');
    await helpModal.trigger('click');

    expect(mockUseNavigationBar.toogleModal).toHaveBeenCalledTimes(1);
  });
});
