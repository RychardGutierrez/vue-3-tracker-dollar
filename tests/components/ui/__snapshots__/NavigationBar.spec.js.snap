// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NavigationBar component should to match with snapshot 1`] = `
"<header class=\\"sticky top-0 bg-white shadow-sm\\">
  <nav class=\\"container flex flex-col sm:flex-row items-center gap-4 text-black py-6 px-0\\">
    <router-link-stub to=\\"[object Object]\\" replace=\\"false\\" custom=\\"false\\" ariacurrentvalue=\\"page\\"></router-link-stub>
    <div class=\\"flex gap-5 flex-1 justify-end\\">
      <!--v-if--><img src=\\"help-icon\\" alt=\\"Help\\" class=\\"w-9 hover:shadow-xl rounded-full duration-150 cursor-pointer\\">
    </div>
  </nav>
</header>
<help-modal-stub modalactive=\\"false\\"></help-modal-stub>"
`;
